gammapy/visualization/datasets.py

```diff
@@ -0,0 +1,167 @@
+    from matplotlib.patches import CirclePolygon, Patch
+
+    if ax is None:
+        ax = plt.subplot(projection=datasets[0].counts_off.geom.wcs)
+
+    legend = legend or legend is None and len(datasets) <= 10
+    legend_kwargs = legend_kwargs or {}
+    handles, labels = [], []
+
+    prop_cycle = kwargs.pop("prop_cycle", plt.rcParams["axes.prop_cycle"])
+
+    for props, dataset in zip(prop_cycle(), datasets):
+        plot_kwargs = kwargs.copy()
+        plot_kwargs["facecolor"] = "None"
+        plot_kwargs.setdefault("edgecolor")
+        plot_kwargs.update(props)
+
+        dataset.counts_off.plot_region(ax, **plot_kwargs)
+
+        # create proxy artist for the custom legend
+        if legend:
+            handle = Patch(**plot_kwargs)
+            handles.append(handle)
+            labels.append(dataset.name)
+
+    if legend:
+        legend = ax.get_legend()
+        if legend:
+            handles = legend.legendHandles + handles
+            labels = [text.get_text() for text in legend.texts] + labels
+
+        handles = [(handle, handle) for handle in handles]
+        tuple_handler = HandlerTuple(ndivide=None, pad=0)
+
+        def patch_func(
+            legend, orig_handle, xdescent, ydescent, width, height, fontsize
+        ):
+            radius = width / 2
+            return CirclePolygon((radius - xdescent, height / 2 - ydescent), radius)
+
+        patch_handler = HandlerPatch(patch_func)
+
+        legend_kwargs.setdefault("handletextpad", 0.5)
+        legend_kwargs["handler_map"] = {Patch: patch_handler, tuple: tuple_handler}
+        ax.legend(handles, labels, **legend_kwargs)
+
+    return ax
+
+
+def plot_npred_signal(
+    dataset,
+    ax=None,
+    model_names=None,
+    region=None,
+    **kwargs,
+):
+    """
+    Plot the energy distribution of predicted counts of a selection of models assigned to a dataset.
+    Plot the energy distribution of predicted counts of the background and of the sum of all the considered models
+    on the same axis.
+
+    Parameters
+    ----------
+    dataset : an instance of `~gammapy.datasets.dataset`
+        The dataset from which to plot the npred_signal.
+    ax : `~matplotlib.axes.Axes`
+        Axis object to plot on.
+    model_names : list of str
+        The list of models for which the npred_signal is plotted. Default is None.
+        If None, all models are considered.
+    region: `~regions.Region` or `~astropy.coordinates.SkyCoord`
+        Region used to reproject predicted counts. Default is None.
+        If None, use the full dataset geometry.
+    **kwargs : dict
+        Keyword arguments to pass to `~gammapy.maps.RegionNDMap.plot`.
+
+
+    Returns
+    -------
+
+    axes : `~matplotlib.axes.Axes`
+        Axis object
+    """
+
+    npred_stack = dataset.npred_signal(
+        model_names=model_names, stack=True
+    ).to_region_nd_map(region)
+    npred_not_stack = dataset.npred_signal(
```

Comment:
Isn't it simpler to first compute `npred` and then stack it with a `sum_over_axes`?